mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -286,6 +286,24 @@ static void correctParameters() {
   }
 }
 
+static void verifyLayout() {
+  std::string filterLayoutValue = filterLayout.getValue();
+  std::string inputLayoutValue = inputLayout.getValue();
+
+  if (filterLayoutValue.find("yx") == std::string::npos &&
+      filterLayoutValue.find("xy") == std::string::npos) {
+    llvm::errs() << "Unsupported filter layout: disjointed yx!\n";
+    exit(1);
+  }
+
+  if (inputLayoutValue.find("hw") == std::string::npos &&
+      inputLayoutValue.find("wh") == std::string::npos) {
+
+    llvm::errs() << "Unsupported input layout: disjointed hw!\n";
+    exit(1);
+  }
+}
+
```

Comment:
I'm curious why this additional verification in the driver is necessary. It looks like it is redundant. If the layout cannot pass the `Conv2dOp`'s verification stage, I assume it should error out before the first step lowering. Is it not the case?

Reply:
this is to teach @zjing14 that checks can be placed at different places.